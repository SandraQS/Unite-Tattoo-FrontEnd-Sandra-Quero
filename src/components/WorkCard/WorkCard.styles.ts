import { StyleSheet } from "react-native";
import { fontSize, colors } from "../../styles/uniteTatto.styles";

export default StyleSheet.create({
  buttonsEditDelete: {
    position: "absolute",
    right: 0,
    top: 0,
  },
  imageContainer: {
    position: "relative",
  },
  imageWork: {
    flex: 1,
    height: 350,
  },
  styleWork: {
    color: colors.darkGrey,
    fontSize: fontSize.tittleH3,
    fontWeight: "800",
    textTransform: "uppercase",
  },
  textContainer: {
    padding: 15,
  },
  textWork: {
    color: colors.darkGrey,
    fontSize: fontSize.text,
    fontWeight: "200",
  },
  titleWork: {
    color: colors.black,
    fontSize: fontSize.tittleH2,
    fontWeight: "900",
    textTransform: "uppercase",
  },
  workContainer: {
    borderColor: colors.mediumBrown,
    borderWidth: 1,
    margin: 10,
    position: "relative",
  },
});
