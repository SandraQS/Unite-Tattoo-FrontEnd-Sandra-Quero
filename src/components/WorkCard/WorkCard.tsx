import React, { useState } from "react";
import { Image, Text, TouchableOpacity, View } from "react-native";
import { useWorks } from "../../hooks/useWorks";
import { IWorkProps } from "../../types/interfacesComponent";
import { EditDelete } from "../EditDelete/EditDelete";
import styles from "./WorkCard.styles";

const WorkCard = ({ work }: IWorkProps) => {
  const [longPress, setLongPress] = useState(false);
  const { deleteWork, editWork } = useWorks();

  const showButtons = () => {
    setLongPress(!longPress);
  };
  const onPressWork = () => {
    setLongPress(false);
  };

  const deletePress = () => {
    deleteWork(work.id);
    setLongPress(false);
  };

  const editPress = () => {
    editWork(work);
    setLongPress(false);
  };

  return (
    <TouchableOpacity
      onPress={onPressWork}
      onLongPress={showButtons}
      activeOpacity={0.6}
    >
      <View style={styles.workContainer}>
        <View style={styles.imageContainer}>
          <Image source={{ uri: work.image }} style={styles.imageWork} />
        </View>

        <View style={styles.textContainer}>
          <Text style={styles.titleWork}>{work.tittle}</Text>
          <Text style={styles.styleWork}>{work.tattooStyles}</Text>
          <Text style={styles.textWork}>{work.description}</Text>
        </View>

        <View style={styles.buttonsEditDelete}>
          {longPress && (
            <EditDelete functionGoEdit={editPress} pressDelete={deletePress} />
          )}
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default WorkCard;
