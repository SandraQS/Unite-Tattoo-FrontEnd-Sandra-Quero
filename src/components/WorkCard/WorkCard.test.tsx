import { render } from "@testing-library/react-native";
import React from "react";
import WorkCard from "./WorkCard";

describe("Given a WorkCard component", () => {
  describe("When it is rendered", () => {
    test("Then it should show match the last snapshot", () => {
      const work: any = {
        tittle: "Lobo",
        tattooArtist: "Javi Wolff",
        description:
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed felis magna, volutpat eget nulla in, consequat laoreet nisl. Integer faucibus elit sapien, sit amet volutpat libero congue non. Ut est erat, feugiat in feugiat sit amet, interdum at risus. Cras fringilla congue ipsum, vel vulputate diam fermentum vel. Mauris accumsan neque eget nisl consequat suscipit. Fusce aliquet mi quis mi condimentum egestas. Nunc gravida tempor commodo. Morbi gravida lectus quis augue faucibus ultricies. Fusce mi dui, interdum vitae scelerisque et, cursus id nisi.",
        tattooStyles: "Acuarela",
        likes: 0,
        image:
          "https://storage.googleapis.com/unite-tattoo.appspot.com/Acuarela-1638119881947-.png",
        collectionWork: "61a37f78b0e47a2b50e4635b",
        id: "61a3b9ca4cc19250ca826626",
      };

      const screen = render(<WorkCard key={work.id} work={work} />);

      expect(screen).toMatchSnapshot();
    });
  });
});
