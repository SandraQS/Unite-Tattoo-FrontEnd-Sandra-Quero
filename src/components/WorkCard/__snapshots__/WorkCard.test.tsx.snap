// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a WorkCard component When it is rendered Then it should show match the last snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#A19178",
        "borderWidth": 1,
        "margin": 10,
        "position": "relative",
      }
    }
  >
    <View
      style={
        Object {
          "position": "relative",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://storage.googleapis.com/unite-tattoo.appspot.com/Acuarela-1638119881947-.png",
          }
        }
        style={
          Object {
            "flex": 1,
            "height": 350,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "padding": 15,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#141414",
            "fontSize": 25,
            "fontWeight": "900",
            "textTransform": "uppercase",
          }
        }
      >
        Lobo
      </Text>
      <Text
        style={
          Object {
            "color": "#787878",
            "fontSize": 16,
            "fontWeight": "800",
            "textTransform": "uppercase",
          }
        }
      >
        Acuarela
      </Text>
      <Text
        style={
          Object {
            "color": "#787878",
            "fontSize": 18,
            "fontWeight": "200",
          }
        }
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed felis magna, volutpat eget nulla in, consequat laoreet nisl. Integer faucibus elit sapien, sit amet volutpat libero congue non. Ut est erat, feugiat in feugiat sit amet, interdum at risus. Cras fringilla congue ipsum, vel vulputate diam fermentum vel. Mauris accumsan neque eget nisl consequat suscipit. Fusce aliquet mi quis mi condimentum egestas. Nunc gravida tempor commodo. Morbi gravida lectus quis augue faucibus ultricies. Fusce mi dui, interdum vitae scelerisque et, cursus id nisi.
      </Text>
    </View>
    <View
      style={
        Object {
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
